### DISCUSSION TITLE
50 million + products in Saleor, PostgreSQL

### DISCUSSION THREAD

**Discussion by al3wb** [saleor/saleor#13490] [Q&A]
Hello!

We are creating a store of affiliate products. Approximately 1800 categories (20 large and many nested subsidiaries). About 50 million products at the start (in the future will be more).

Will Saleor and PostgreSQL be able to handle such a number of products? Can Saleor work with a sharded PostgreSQL database?

We have a lot of servers. Most likely for such a number of products need to split Saleor and the database, but how to do it is not clear.

I don't want to write a self-written system from scratch, I would like to use the ready functionality of Saleor.

**Comment by timuric** [comment]
Hey @al3wb yes, Saleor can handle such scale, feel free to reach out to hello@saleor.io if you are interested to talk to our support assistance
